Harvard students replication.R

```diff
@@ -0,0 +1,349 @@
+##Replication Code for "The Unholy Trinity: Immigration, Unemployment and Extremism in Europe, 1980-2002"
+##Jeffrey Javed and Noam Gidron
+##May 8, 2011
+
+##Load libraries.
+
+library(Zelig)
+library(lme4)
+library(MASS)
+library(nlme)
+library(plm)
+library(xtable)
+
+##Load Arzheimer (2009) data.
+
+data <- as.data.frame(read.delim("nonimp.tab.tsv"))
+
+##Center variables (done as according to the author).
+
+salienzmean.c <- data$salienzmean - 3.84568
+rvar.c <- data$rvar - 21.75423
+
+##Create country dummies.
+
+country <- factor(data$sortcountry, labels=c("AT", "BE","DE-E","DE-W","DK","ES","FI","FR","GR","IT","LU","NL","NO","PT","SE"))
+
+##Run with PQL.
+ 
+out.pql <- glmmPQL(rexvote ~ male + age1 + age2 + age4 + mye1 + mye2 + farmerown + worker + retired + unemployed + zlrs + euschlecht + zsatisdmo + disp + lfed1 + zasylumseekers + zsur + zasylumseekers:zsur + zreplacementrate + zsur:zreplacementrate + zasylumseekers:zreplacementrate + rmax + salienzmean.c + rvar.c + rvar.c:salienzmean.c + country - 1, random=~1|kontext, family=binomial(link="logit"), data=data, verbose=TRUE)
+
+summary(out.pql)
+
+####Simulations for the Original Model with Centered Variables##
+
+##Simulate betas for centered data set.
+
+set.seed(02138)
+simbetas <- mvrnorm(n=10000, mu=out.pql$coef$fixed, Sigma=out.pql$var)
+
+##Create matrix for all variables.
+
+mat1 <- cbind(data$rexvote, data$male, data$age1, data$age2, data$age4, data$mye1, data$mye2, data$farmerown, 

```

Comment:
wow. this makes sense because you're selecting exactly what you want, but know that you could also use the dplyr select tool. If you use: data %>% select(-rexvote), it'll select everything but "rexvote."